lib/versioner/systemtrash_darwin.go

```diff
@@ -0,0 +1,87 @@
+// Copyright (C) 2016 The Syncthing Authors.
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at http://mozilla.org/MPL/2.0/.
+
+//+build darwin,cgo
+
+package versioner
+
+/*
+#cgo CFLAGS: -x objective-c
+#cgo LDFLAGS: -framework Foundation
+#import <stdlib.h>
+#import <Foundation/Foundation.h>
+
+const char *trashFile(char *s) {
+	NSAutoreleasePool *pool = [NSAutoreleasePool new];
```

Comment:
Ah, no, you need the `AutoreleasePool` for the memory allocated for that `NSError`, so it needs to be around (and for the `fileStr`, since you created it with `stringWithUTF8String`).

Can you get away with use `@autorelease`, i.e.

```objective-c
    @autorelease  {
        NSString *fileStr = [NSString stringWithUTF8String:s];
	NSURL *fileURL = [NSURL fileURLWithPath:fileStr];
	NSError *err = nil;
	BOOL result = [[NSFileManager defaultManager] trashItemAtURL:fileURL resultingItemURL:nil error:&err];
	if (!result) {
		char *str = strdup([[err localizedDescription] UTF8String]);
		return str;
	}
	return nil;
    }
```